lib/recma-jsx-rewrite.js

```diff
@@ -3,14 +3,30 @@ import {walk} from 'estree-walker'
 import {analyze} from 'periscopic'
 import u from 'unist-builder'
 
-// A plugin that rewrites JSX in functions to accept components as
-// `props.components` (when the function is called `MDXContent`), or from
-// a provider (if there is one).
-// It also makes sure that any undefined components are defined: either from
-// received components or as a function that throws an error.
+/**
+ * @typedef {import("periscopic").Node} Node
```

Comment:
I tried using using `import("estree").Node` the issue comes in on line 32.
`analyze` uses the definition from line 60 https://unpkg.com/browse/periscopic@3.0.0/types/index.d.ts
Which is not compatible with the `import("estree").Node` that can be directly imported, either due to an override or due to a version conflict.

Reply:
The results from the use of periscopic here is indeed only an identifier node (say, a strong node for mdast).
But the rest of the tree is a complete estree node.

Anyway, I don’t know enough about TS to have a meaningful idea here, but I’d guess that using the whole estree types here should be possible, and “narrowing” it to identifiers for the result of periscopic could also work?

Reply:
ohh, I guess their Node is too specific compared to an estree Node? 🤔 Weird.

Reply:
I'm a bit confused by it, https://unpkg.com/browse/periscopic@3.0.0/types/index.d.ts line 1 indicates it should be the same as `estree`, but line 60 appears to make it more specific/narrow... :thinking:

Reply:
And would actually using `estree` for the other places (https://github.com/wooorm/xdm/pull/2/files/d347c39c923aa71cfd19e6801177f0f22d0e6877#diff-ca1bb3ad6385be868a6efa965db2bbb1da4fedd201ae4fc1f11e3d8923d95115R41-R48) help?

Reply:
It may not be a bad idea to use `estree.Node` in other places.
It wouldn't fix this particular issue, which is a conflict between the signature of the method and a function call inside the method.

Reply:
smoothed over this with an ignore for consistency of using estree (b466ab6)